privaterelay/views.py

```diff
@@ -140,15 +141,21 @@ def waitlist(request):
 def fxa_rp_events(request):
     jwt = _parse_jwt_from_request(request)
     authentic_jwt = _authenticate_fxa_jwt(jwt)
+    event_keys = _get_event_keys_from_jwt(authentic_jwt)
     try:
         social_account = _get_account_from_jwt(authentic_jwt)
     except SocialAccount.DoesNotExist as e:
         # we received an FXA event for an FXA not in our DB;
-        # capture the exception in sentry, but don't error, or FXA will retry
+        # If the user joined the waitlist, we didn't create a social_account,
+        # but we did create an Invitations object. Delete it if necessary.
+        for event_key in event_keys:
+            if (event_key == FXA_DELETE_EVENT):
+                _delete_invitation(authentic_jwt)
```

Comment:
Doesn't social_account get created if the user logs in to our app successfully (can access the private-relay create address page)? If so we should also delete the social account.

Reply:
Yup, but this call is inside `except SocialAccount.DoesNotExist`. If the user has a social account object, we process the delete event in `_handle_fxa_delete` below here.